frontend/Auth/Login.tsx

```diff
@@ -1,9 +1,167 @@
-import { Button, Form } from "semantic-ui-react";
+/*
+[x] new frontend route /login. 
+[x] contain a card with inputs for email, password, and a submit button.
+
+[x] When the form submits, it should make a request to the backend to login, possibly sending the password as a base64 encoded string.
+
+[x] If the backend responds with a 200, it can be assumed that a cookie was set and the user is now logged in. At that time, the user should be redirected back to the home page.
+*/
+
+import React, { useState } from "react";
+import { useHistory, Link } from "react-router-dom";
+import { Button, Form, Card } from "semantic-ui-react";
+import { RouterProps } from "react-router";
+import { useMutation } from "react-query";
+import { flaxFetch } from "../Utils/flaxFetch";
+import { unary } from "lodash-es";
+
+export function Login(props: RouterProps) {
+  const [loginFormData, setLoginFormData] = useState<LoginFormData>({
+    username: "",
+    password: "",
+  });
+
+  const history = useHistory();
+
+  const submitMutation = useMutation<
+    LoginResultData,
+    Error,
+    React.FormEvent<HTMLFormElement>
+  >(
+    (evt) => {
+      evt.preventDefault();
+      let requestPromise = flaxFetch<LoginResultData>(`/login`, {
+        method: "POST",
+        body: {
+          username: loginFormData.username,
+          password: loginFormData.password,
+        },
+      });
+      return requestPromise;
+    },
+    {
+      onSuccess: async (data, variables, context) => {
+        history.push("/home");
+      },
+      onError: (error, variables, context) => {
+        console.log(error);
```

Comment:
Ideally this would report the error to an error system like Sentry so that we'd be aware of it when a user experiences it. But that's a different task for a different day

Reply:
I will add this to a list of items to revisit.